### PR TITLE
remove dependency on vue-template-compiler

### PR DESCRIPTION
**Pull request by HerringtonDarkholme**
Fix https://github.com/octref/vetur/issues/250

### CODE REVIEW

File: server/src/modes/embeddedSupport.ts [hunk 1]
```diff
@@ -218,65 +216,14 @@ function getLanguageAtPosition (document: TextDocument, regions: EmbeddedRegion[
...
 
 function getEmbeddedDocument (document: TextDocument, contents: EmbeddedRegion[], languageId: string): TextDocument {
-  let currentPos = 0;
   let oldContent = document.getText();
   let result = '';
-  let lastSuffix = '';
   for (let c of contents) {
     if (c.languageId === languageId) {
-      result = substituteWithWhitespace(result, currentPos, c.start, oldContent, lastSuffix, getPrefix(c));
+      result = oldContent.substring(0, c.start).replace(/./g, ' ');
```
Comment (HerringtonDarkholme) [inline server/src/modes/embeddedSupport.ts]: We don't need `prefix` nor `suffix` in current code base. It is for inline style attribute.

File: server/src/modes/script/javascript.ts [hunk 2]
```diff
@@ -151,9 +151,9 @@ export function getJavascriptMode (documentRegions: LanguageModelCache<VueDocume
         };
       }
       const normalizedFileFsPath = getNormalizedFileFsPath(fileName);
-      let doc = docs.get(normalizedFileFsPath);
+      let doc = scriptDocs.get(normalizedFileFsPath);
       let text = doc ? doc.getText() : (ts.sys.readFile(normalizedFileFsPath) || '');
-      if (isVue(fileName)) {
+      if (!doc && isVue(fileName)) {
```
Comment (HerringtonDarkholme) [inline server/src/modes/script/javascript.ts]: We only need to `parseVue` for text read from disk. If doc is from `scriptDocs`, it is already plain script file.

File: server/src/modes/embeddedSupport.ts [hunk 3]
```diff
@@ -175,7 +173,7 @@ function getLanguageRanges (document: TextDocument, regions: EmbeddedRegion[], r
           start: startPos,
           end: endPos,
           languageId: region.languageId,
-          attributeValue: region.attributeValue
+          // attributeValue: region.attributeValue
```
Comment (octref) [inline server/src/modes/embeddedSupport.ts]: Maybe just remove it.

**Comment by HerringtonDarkholme** [comment]
Travis works well!

**Pull request merged**